src/Aether/Devices/Drivers/Sgp4x.cs

```diff
@@ -0,0 +1,162 @@
+
+            // Check result status (Most significant byte, ignore least significant non-crc byte)
+            // 0xD4 = All tests passed
+            // 0x4B = One or more test failed
+            return testResultWithCRC[0] == 0xD4;
+        }
+
+        /// <summary>
+        /// Gets the raw VOC measurements from the sensor.
+        /// </summary>
+        /// <param name="relativeHumidityValue">The relative humidity value expressed as a percentage.</param>
+        /// <param name="temperatureValue">The temperature value expressed in degrees celsius.</param>
+        /// <param name="disableHotPlate">If true, the hot plate will be disabled after measurements are taken.</param>
+        /// <returns>The raw VOC measurement value from the sensor. If an error occurred, it will be <see langword="null"/>.</returns>
+        /// <remarks>If default values are supplied, humidity compensation will be disabled.</remarks>
+        public ushort? GetVOCRawMeasure(ushort relativeHumidityValue = 50, short temperatureValue = 25, bool disableHotPlate = true)
+        {
+            if (relativeHumidityValue > 100)
+            {
+                throw new ArgumentOutOfRangeException(nameof(relativeHumidityValue), relativeHumidityValue, "A relative humidity percentage value must be between 0 and 100");
+            }
+
+            if (temperatureValue < -45 || temperatureValue > 130)
+            {
+                throw new ArgumentOutOfRangeException(nameof(temperatureValue), temperatureValue, "A temperature value must be between -45 and 130");
+            }
+
+            Span<byte> writeBuffer = stackalloc byte[10];
+            Span<byte> readBuffer = stackalloc byte[3];
+
+            // Write start measure VOC raw
+            writeBuffer[0] = 0x26;
+            writeBuffer[1] = 0x0F;
+
+            // Write default humidity value + CRC (0x80, 0x00, [CRC], 0xA2)
+            Sensirion.WriteUInt16BigEndianAndCRC8(writeBuffer.Slice(2, 3), (ushort)(relativeHumidityValue * 65535 / 100));
+            writeBuffer[5] = 0xA2;
+
+            // Write default temperature value + CRC (0x66, 0x66, [CRC], 0x93)
+            Sensirion.WriteUInt16BigEndianAndCRC8(writeBuffer.Slice(6, 3), (ushort)((temperatureValue + 45) * 65535 / 175));
+            writeBuffer[9] = 0x93;
+
+            // Transmit command
+            _device.Write(writeBuffer);
+
+            Thread.Sleep(30);
+
+            _device.Read(readBuffer);
+
+            // Disable hot plate if instructed
+            if (disableHotPlate)
+            {
+                DisableHotPlate();
+            }
+
+            // Read the results and validate CRC
+            return Sensirion.ReadUInt16BigEndianAndCRC8(readBuffer);
+        }
+
+        /// <summary>
+        /// Disables the hot plate on the device.
+        /// </summary>
+        public void DisableHotPlate()
+        {
+            _device.WriteByte(0x36);
+            _device.WriteByte(0x15);
```

Comment:
This will generate different protocol on the wire than `Write(span)`. Is it correct? In `GetVOCRawMeasure` two command bytes are being written together in one span.

Reply:
Changed to Write(span)